common/djangoapps/track/shim.py

```diff
@@ -119,15 +121,40 @@ def __call__(self, event):
 
             del payload['module_id']
 
+        if 'context' not in event:
+            return
+
+        context = event['context']
+
+        #For the iOS build that is returning a +30 for back skip 30
+        if context['application']['version'] == "1.0.02" and \
+                        context['application']['name'] == "edx.mobileapp.iOS":
+            if 'requested_skip_interval' in payload and 'seek_type' in payload:
+                if payload['requested_skip_interval'] == 30 and \
+                        payload['seek_type'] == "skip":
+                    payload['requested_skip_interval'] = -30
+
+        #For the Android build that isn't distinguishing between skip and seek
+        if 'requested_skip_interval' in payload:
+            if payload['requested_skip_interval'] != -30:
+                if 'seek_type' in payload:
+                    payload['seek_type'] = 'slide'
+
+        #For converting seek_type to seek and skip|slide to onSlideSeek|onSkipSeek
+        if 'seek_type' in payload:
```

Comment:
As we discussed, suggestion to mode this block of code above the Android block above (and you'll need to replace 'seek_type' with 'OnSlideSleek' in the Android block).